packages/public/server/src/module/Entity/src/Controller/RepositoryController.php

```diff
@@ -437,24 +438,19 @@ protected function handleAddRevisionPost(
                 'Your revision has been saved and is available'
             );
             $route = 'entity/page';
+            $hash = '#revision-saved-accepted';
         } else {
             $successMessage = $translator->translate(
                 'Your revision has been saved, it will be available once it gets approved'
             );
             $route = 'entity/repository/history';
+            $hash = '#revision-saved';
         }
         $this->getEntityManager()->flush();
         $this->flashMessenger()->addSuccessMessage($successMessage);
 
-        if ($this->getRequest()->isXmlHttpRequest()) {
-            return $this->plugin('url')->fromRoute($route, [
-                'entity' => $entity->getId(),
-            ]);
-        }
-
-        return $this->redirect()->toRoute($route, [
-            'entity' => $entity->getId(),
-        ]);
+        $entityId = ['entity' => $entity->getId()];
```

Comment:
Why is this an extra variable? Isn't needed and additionally confusing because `$entityId` is not the entity's id.